File: go/maybehug.go
package maybehug

// Cutie is someone to hug (or not)
type Cutie interface {
	RespondsToHug() bool
}

// MaybeHug maybe hugs a cutie
func MaybeHug(c Cutie) string {
	if c.RespondsToHug() {
		return "Hug!"
	}
	return "Empathy!"
}

File: typescript/MaybeHug.ts
interface Cutie {
    name: string;
    acceptsHugs: boolean;
}

export class MaybeHug {
    constructor(private cutie: Cutie) {
        if (cutie === null || cutie === undefined) {
            throw new TypeError('no cutie to maybe hug');
        }
        this.cutie = cutie;
    }
    hug () {
        return this.cutie.acceptsHugs ? 'HUG!' : 'Empathy!';
    }
    love() {
        return `I love ${this.cutie.name}`;
    }
}
